route/v1/types.go

```diff
@@ -139,8 +139,90 @@ type RouteSpec struct {
 	// +kubebuilder:validation:Enum=None;Subdomain;""
 	// +kubebuilder:default=None
 	WildcardPolicy WildcardPolicyType `json:"wildcardPolicy,omitempty" protobuf:"bytes,7,opt,name=wildcardPolicy"`
+
+	// httpHeaders defines policy for HTTP headers.
+	//
+	// If this field is empty, the default values are used.
+	//
+	// +optional
+	HTTPHeaders *RouteHTTPHeaders `json:"httpHeaders,omitempty" protobuf:"bytes,9,opt,name=httpHeaders"`
```

Comment:
Why use a pointer type?

Reply:
@JoelSpeed do you think we shall remove pointer type ?

Reply:
BTW in Ingress Controller API we use a pointer 
	// httpHeaders defines policy for HTTP headers.
	//
	// If this field is empty, the default values are used.
	//
	// +optional
	HTTPHeaders *IngressControllerHTTPHeaders `json:"httpHeaders,omitempty"`

Reply:
I don't think a pointer is needed here, I don't see anything within the struct that would make me think we need this to be a pointer

Reply:
I think I will leave it as a pointer as we already use a pointer 	HTTPHeaders *IngressControllerHTTPHeaders `json:"httpHeaders,omitempty"` in the ingress controller API.

Reply:
Repeating the sins of the past is not a valid excuse when adding new code to the API repo, unless there's some technical reason this needs to be a pointer, it should be moved to a reference else you're adding complexity for no reason

Reply:
Yes i updated the route api to remove the sin 😂